packages/core/src/components/ui/Truncate/Truncate.js

```diff
@@ -1,12 +1,20 @@
 import cx from 'classnames'
+import _truncate from 'lodash/truncate'
 import PropTypes from 'prop-types'
 import React, { forwardRef } from 'react'
 
 import * as Classes from '../../../common/classes'
 import Tooltip from '../Tooltip'
 
 // eslint-disable-next-line prefer-arrow-callback
-const Truncate = forwardRef(function Truncate({ className, children, ...rest }, ref) {
+const Truncate = forwardRef(function Truncate({
+  limit, className, children, ...rest
+}, ref) {
+  const shortedText = _truncate(children, {
+    length: limit,
+    omission: '...',
+  })
```

Comment:
it handles and shorts text by following css, so we shouldn't add JS-code to show ellipsis

 white-space: nowrap;
  text-overflow: ellipsis;
  overflow: hidden;